app/models/os_major_version.rb

```diff
@@ -45,14 +49,25 @@ def self.aggregated_cast(column)
                         inverse_of: false, dependent: :restrict_with_exception
 
   has_many :profiles, lambda {
+    # List the profiles in use for the current user
+    profiles_in_use = User.current.account.profiles.joins(PROFILE_BM_JOIN.join_sources)
+                          .select(arel_table[:ref_id], OS_MAJOR_VERSION, PROFILE_IN_USE)
+
+    # Canonical profiles joined with profiles in use and grouped by supported benchmark versions
     supported_profiles = canonical.where(upstream: false)
-                                  .joins(:benchmark)
-                                  .select(PROFILE_LAST_ID, PROFILE_BM_VERSIONS, OS_MAJOR_VERSION)
-                                  .group(:ref_id, 'os_major_version')
+                                  .joins(PROFILE_BM_JOIN.join_sources)
+                                  .joins("LEFT OUTER JOIN (#{profiles_in_use.to_sql}) sq ON
+                                    sq.ref_id = profiles.ref_id AND
+                                    os_major_version = sq.os_major_version")
+                                  .select(PROFILE_LAST_ID, PROFILE_BM_VERSIONS, OS_MAJOR_VERSION, 'sq.in_use')
+                                  .group(:ref_id, Xccdf::Benchmark.arel_table[:ref_id], 'in_use')
 
+    # Rails cannot consume grouped results as proper ActiveRecord classes, it has to be joined with itself
     canonical.where(upstream: false)
              .joins("INNER JOIN (#{supported_profiles.to_sql}) t ON t.id = profiles.id")
-             .select('"profiles".*, "t"."bm_versions" AS "bm_versions", "t"."os_major_version" AS "os_major_version"')
+             .select('"profiles".*, "t"."bm_versions" AS "bm_versions"',
+                     '"t"."os_major_version" AS "os_major_version"',
+                     '"t"."in_use" AS "in_use"')
```

Comment:
What value would `in_use` have for profiles that are NOT in use? Is it `NULL`?

Reply:
Yes

Reply:
And it gets converted to boolean in GQL [here](https://github.com/RedHatInsights/compliance-backend/pull/1445/files#diff-55bef2bb2e7b33ab10c18422e60ddb05e02505244e59107f8ce099417e74b6d9R113)

Reply:
I see. Wouldn't it be safer to do `COALESCE(t.in_use, FALSE)`? Now it is pushed downwards to GraphQL.

Reply:
Good idea!